config/application.rb

```diff
@@ -117,18 +117,17 @@ class Application < Rails::Application
           next if source == IdentityConfig.store.domain_name
 
           redirect_uris = Rails.cache.fetch(
-            'all_service_provider_redirect_uris',
+            'all_service_provider_redirect_uris_cors',
             expires_in: IdentityConfig.store.all_redirect_uris_cache_duration_minutes.minutes,
           ) do
-            ServiceProvider.pluck(:redirect_uris).flatten.compact
+            ServiceProvider.pluck(:redirect_uris).flatten.compact.map do |uri|
+              protocol, domain_path = uri.split('//', 2)
+              domain, _path = domain_path&.split('/')
```

Comment:
Micro-optimization: Could pass the second argument here too, especially if we're not using anything but the first segment?

```suggestion
              domain, _path = domain_path&.split('/', 2)
```